### ISSUE TITLE
Cannot create Input Group in MythTV-Setup

### ISSUE THREAD

**Issue by davepearson1628** [MythTV/mythtv#698]
Using mythtv v32 on Ubuntu 22.10

I was trying to set up an input group for 2 External Recorders (Same device, different profile) so that only one can be used at a time.

It should be possible to do this in Input Connections, but the new Input Group does not get saved, so Inputs cannot be added to it.

Current work around is to manually update the inputgroup table.

Regards,
Dave

**Comment by kmdewaal** [comment]
Can reproduce this on master but also on fixes/31.

**Comment by kmdewaal** [comment]
Looks like commenting out line 1552 in cardutil.cpp 
```
    // Delete any unused input groups
    CardUtil::UnlinkInputGroup(0,0);

```
does fix the problem.
This line is introduced in commit 39b850f1a922c5e51921c135110193389ef11f5b on  Jul 9, 2022, and  it is the last action of the InputSetMaxRecordings.

This is what I think that happens.
If you have created an InputGroup then it is initially created in database table inputgroup with the cardinputid set to 0. Later then a new record with the correct cardinputid is created and the temporary one is deleted.
Defining an input group for the first capturecard works OK.
When you select that input group for the second capture card then on the "Save and Exit"  the InputSetMaxRecordings is called and at the end it does the UnlinkInputGroup. This then deletes the unused input groups for all capture cards including the temporary one for the second capture card that was not yet finalized.

@jpoet please comment

**Comment by dengelintz** [comment]
Klaas, that commit you referenced is definitely wrong.  There's a good chance I reviewed it for John.  If I did, I certainly missed it.  The change to CardInput::Save() should be reverted.  Removing the CardUtil::UnlinkInputGroup(0,0) from CardUtil::InputSetMaxRecordings() should also be done.  It's probably better, though, to move the whole CloneCard() loops bit to a separate helper function as it's likely needed by any API methods which modfy the capturecard table.
